eos/eos.hpp

```diff
@@ -448,6 +448,137 @@ template <typename T> class RelativisticEOS {
+    const Real alpha {this->alpha_rho_temperature(rho, temperature, lambda)};
+    return
+      t_.InternalEnergyFromDensityTemperature(alpha*rho, temperature, lambda);
+  }
+
+  PORTABLE_FUNCTION
+  Real PressureFromDensityInternalEnergy(const Real rho, const Real sie,
+                                         Real *lambda = nullptr) const {
+    return std::get<1>(this->alpha_impl(&t_.PressureFromDensityInternalEnergy,
+					rho, sie, lambda));
+  }
+
+  PORTABLE_FUNCTION
+  Real SpecificHeatFromDensityInternalEnergy(const Real rho, const Real sie,
+                                             Real *lambda = nullptr) const {
+    const Real alpha {this->alpha_rho_sie(rho, sie, lambda)};
+    return t_.SpecificHeatFromDensityInternalEnergy(alpha*rho, sie, lambda);
+  }
+
+  PORTABLE_FUNCTION
+  Real BulkModulusFromDensityInternalEnergy(const Real rho, const Real sie,
+                                            Real *lambda = nullptr) const {
+    const Real alpha {this->alpha_rho_sie(rho, sie, lambda)};
+    return t_.BulkModulusFromDensityInternalEnergy(alpha*rho, sie, lambda);
+  }
+
+  PORTABLE_FUNCTION
+  Real GruneisenParamFromDensityInternalEnergy(const Real rho, const Real sie,
+                                               Real *lambda = nullptr) const {
+    const Real alpha {this->alpha_rho_sie(rho, sie, lambda)};
+    return t_.GruneisenParamFromDensityInternalEnergy(alpha*rho, sie, lambda);
+  }
+
+  PORTABLE_FUNCTION
+  Real PressureFromDensityTemperature(const Real rho, const Real temperature,
+                                      Real *lambda = nullptr) const {
+    return std::get<1>(this->alpha_impl(&t_.PressureFromDensityTemperature,
+					rho, temperature, lambda));
+  }
+
+  PORTABLE_FUNCTION
+  Real SpecificHeatFromDensityTemperature(const Real rho,
+                                          const Real temperature,
+                                          Real *lambda = nullptr) const {
+    const Real alpha {this->alpha_rho_temperature(rho, temperature, lambda)};
+    return
+      t_.SpecificHeatFromDensityTemperature(alpha*rho, temperature, lambda);
+  }
+
+  PORTABLE_FUNCTION
+  Real BulkModulusFromDensityTemperature(const Real rho, const Real temperature,
+                                         Real *lambda = nullptr) const {
+    const Real alpha {this->alpha_rho_temperature(rho, temperature, lambda)};
+    return t_.BulkModulusFromDensityTemperature(alpha*rho, temperature, lambda);
+  }
+
+  PORTABLE_FUNCTION
+  Real GruneisenParamFromDensityTemperature(const Real rho,
+                                            const Real temperature,
+                                            Real *lambda = nullptr) const {
+    const Real alpha {this->alpha_rho_temperature(rho, temperature, lambda)};
+    return
+      t_.GruneisenParamFromDensityTemperature(alpha*rho, temperature, lambda);
+  }
+
+private:
+  T t_;
+  Real Pe_, Pc_, de_pc_, de_pe_, P0_, T0_, r0_, alpha0_;
```

Comment:
Do some of these quantities change with space? If so, should we build them into `lambda` now rather than wait?

Reply:
My understanding, albeit poor, is that these quantities are fixed for a given material. The other issue is that the necessary parameters are going to be P-alpha model dependent. There have been discussions about how extensible the ramp implementation should be and those are up for debate and I'd be happy to discuss further.

Reply:
I was just confused... I thought there was some spatially dependent stuff? Is that not the case?

Reply:
The spatial varying values are going to be passed in via lambda, but these shouldn't be. We can discuss more offline if needed.